src/code.jl

```diff
@@ -0,0 +1,177 @@
+module Code
+
+export toexpr, Assignment, (←), Let, Func, DestructuredArgs,
+       SetArray, MakeArray, MakeSparseArray, MakeTuple
+
+import ..SymbolicUtils
+import SymbolicUtils: @matchable, Sym, Term, istree, operation, arguments
+
+##== state management ==##
+
+struct NameState
+    symbolify::Set
+    destructed_args::Dict
+end
+NameState() = NameState(Set{Any}(), IdDict())
+
+struct LazyState
+    ref::Ref{Any}
+end
+LazyState() = LazyState(Ref{Any}(nothing))
+
+function Base.get(st::LazyState)
+    s = getfield(st, :ref)[]
+    s === nothing ? getfield(st, :ref)[] = NameState() : s
+end
+
+@inline Base.getproperty(st::LazyState, f::Symbol) = getproperty(get(st), f)
+
+##========================##
+
+toexpr(x) = toexpr(x, LazyState())
+toexpr(s::Sym, st) = nameof(s)
+
+@matchable struct Assignment
+    lhs
+    rhs
+end
+lhs(a::Assignment) = a.lhs
+rhs(a::Assignment) = a.rhs
+
+const (←) = Assignment
```

Comment:
We don't want to force the user to use Unicode. Can we also alias it to `:=`?

Reply:
It is not a binary operator...

Reply:
They can always use `Assignment(a, b)` :-p

Reply:
haha